conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/GramineEnclaveEnvironment.kt

```diff
@@ -0,0 +1,118 @@
+package com.r3.conclave.enclave.internal
+
+import com.r3.conclave.common.EnclaveMode
+import com.r3.conclave.common.internal.*
+import com.r3.conclave.utilities.internal.digest
+import java.nio.ByteBuffer
+
+/**
+ * This class currently implements something like mock attestation (see [MockEnclaveEnvironment]).
+ * TODO: (CON-1178) Implement proper attestation
+ */
+class GramineEnclaveEnvironment(
+        enclaveClass: Class<*>,
+        override val hostInterface: SocketEnclaveHostInterface
+) : EnclaveEnvironment(loadEnclaveProperties(enclaveClass, false), null) {
+    companion object {
+        private fun versionToCpuSvn(num: Int): ByteArray {
+            return digest("SHA-256") {
+                update(ByteBuffer.allocate(2).putShort(num.toShort()).array())
+            }.copyOf(SgxCpuSvn.size)
+        }
+    }
+
+    // TODO: (CON-1178) Implement proper attestation
+    override val enclaveMode = EnclaveMode.SIMULATION
+
+    private val tcbLevel = 1
+
+    private val currentCpuSvn: ByteArray by lazy {
+        versionToCpuSvn(tcbLevel)
+    }
+
+    private val mrenclave: ByteArray by lazy {
+        digest("SHA-256") { update(enclaveClass.name.toByteArray()) }
+    }
+
+    private val mrsigner: ByteArray by lazy {
+        ByteArray(32)
+    }
+
+    private val aesSealingKey by lazy(LazyThreadSafetyMode.NONE) {
+        digest("SHA-256") { update(enclaveClass.name.toByteArray()) }.copyOf(16)
+    }
+
+    // TODO: (CON-1178) Implement proper attestation
+    override fun createReport(
+            targetInfo: ByteCursor<SgxTargetInfo>?,
+            reportData: ByteCursor<SgxReportData>?
+    ): ByteCursor<SgxReport> {
+        val report = Cursor.allocate(SgxReport)
+        val body = report[SgxReport.body]
+        if (reportData != null) {
+            body[SgxReportBody.reportData] = reportData.buffer
+        }
+        body[SgxReportBody.cpuSvn] = ByteBuffer.wrap(currentCpuSvn)
+        body[SgxReportBody.mrenclave] = ByteBuffer.wrap(mrenclave)
+        body[SgxReportBody.mrsigner] = ByteBuffer.wrap(mrsigner)
+        body[SgxReportBody.isvProdId] = productID
+        // Revocation level in the report is 1 based. We subtract 1 from it when reading it back from the report.
+        body[SgxReportBody.isvSvn] = revocationLevel + 1
+        body[SgxReportBody.attributes][SgxAttributes.flags] = SgxEnclaveFlags.DEBUG
```

Comment:
Do you know why is this "DEBUG"?

Reply:
No particular reason. Just seems to make more sense to have it than not.